src/odemis/gui/cont/tabs.py

```diff
@@ -2052,68 +2053,57 @@ def _on_change_project_folder(self, evt):
         ans = box.ShowModal()  # Waits for the window to be closed
         if ans == wx.ID_NO:
             return
+
+        prev_dir = self.conf.pj_last_path
+
         # Generate suggestion for the new project name to show it on the file dialog
-        np = create_projectname(self.conf.pj_last_path, self.conf.pj_ptn, count=self.conf.pj_count)
+        root_dir = os.path.dirname(self.conf.pj_last_path)
+        np = create_projectname(root_dir, self.conf.pj_ptn, count=self.conf.pj_count)
         new_dir = ShowChamberFileDialog(self._tab_panel, np)
-        if new_dir is None:
+        if new_dir is None: # Cancelled
             return
-        # Reset project, clear the data
-        self._reset_project_data()
-        # Get previous project dir from configuration
-        prev_dir = self.conf.pj_last_path
 
-        def check_same_partition(source, destination):
-            """
-            Check if source and destination dirs in the same partition
-            """
-            return os.stat(source).st_dev == os.stat(destination).st_dev
+        logging.debug("Selected project folder %s", new_dir)
 
-        def move_files(source, destination):
-            """
-            Move all files found in source directory to destination
-            """
-            file_names = os.listdir(source)
-            for file_name in file_names:
-                shutil.move(os.path.join(source, file_name), destination)
+        # Three possibilities:
+        # * The folder doesn't exists yet => create it
+        # * The folder already exists and is empty => nothing else to do
+        # * The folder already exists and has files in it => Error (because we might override files)
```

Comment:
Though with the current filedialog, there is no way to choose an existing folder . The Save button will always open the existing folder if it's selected or added on the Name text field.

Reply:
Yes, I'm aware that with the current GTK3 version, it's not possible. Still, I wanted to handle this (theoretical) case, so that if the file selection dialog changes behaviour, Odemis handles it fine too.